Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/MecanumDrive.java

```diff
@@ -215,6 +321,42 @@ public MecanumDrive(HardwareMap hardwareMap, Pose2d pose) {
         FlightRecorder.write("MECANUM_PARAMS", PARAMS);
     }
 
+    /**
+     * moves the slides at a power
+     * @param power positive is slides up
+     */
+    public void moveSlides(double power) {
+        this.slideMotor.setPower(power);
+        // if the power is negative, which is slides going down
+        if (power > 0) {
+            this.returnMotor.setPower(power * Constants.SLIDE_RETURN_UP_MUL);
+        } else {
+            this.returnMotor.setPower(power * Constants.SLIDE_RETURN_POWER_MULTIPLIER + Constants.SLIDE_RETURN_POWER_OFFSET);
+        }
+    }
+
+    /**
+     * move slides until within a position tolerance
+     * @param targetPos the destination position of the slides
+     * @return true if it is NOT yet close enough to target pos
+     */
+    public boolean moveSlidesPreset(int targetPos) {
+        double power = (targetPos - this.slideMotor.getCurrentPosition()) * Constants.SLIDE_P_GAIN;
+        this.moveSlides(clamp(power, -Constants.AUTO_SLIDES_MAX_SPEED, Constants.AUTO_SLIDES_MAX_SPEED));
+        return Math.abs(power) >= Constants.AUTO_SLIDES_PRESET_TOLERANCE;
+    }
+
+    public boolean moveSuspensionArmPreset(int targetPos) {
+        double power = (targetPos - this.motSuspension.getCurrentPosition()) * Constants.SLIDE_P_GAIN;
+        this.motSuspension.setPower(clamp(power, -Constants.AUTO_SLIDES_MAX_SPEED, Constants.AUTO_SLIDES_MAX_SPEED));
+        return Math.abs(power) >= Constants.AUTO_SLIDES_PRESET_TOLERANCE;
+    }
+
+    // clamps value between a minimum and maximum value
+    private static double clamp(double value, double min, double max) {
+        return Math.max(min, Math.min(max, value));
+    }
```

Comment:
Slides functionality is not conceptually related to a Mecanum drive or even Road Runner so does not logically belong in this file. Also, MecanumDrive.java is a file that gets updated periodically with new Road Runner fixes and functionality and so the changes should be minimized simply to make it easier to pick up updates. In other words, you might want to put this functionality elsewhere, not in this file.